src/libs/ajax/leonardo/providers/LeoAppProvider.ts

```diff
@@ -0,0 +1,20 @@
+import { Ajax } from 'src/libs/ajax';
+import { ListAppResponse } from 'src/libs/ajax/leonardo/models/app-models';
+
+export interface LeoAppProvider {
+  listWithoutProject: (args: Record<string, string>, signal?: AbortSignal) => Promise<ListAppResponse[]>;
+  delete: (resourceName: string, appName: string, deleteDisk?: boolean, signal?: AbortSignal) => Promise<void>;
+  pause: (resourceName: string, appName: string, signal?: AbortSignal) => Promise<void>;
+}
+
+export const leoAppProvider: LeoAppProvider = {
+  listWithoutProject: (args: Record<string, string>, signal?: AbortSignal): Promise<ListAppResponse[]> => {
+    return Ajax(signal).Apps.listWithoutProject(args);
+  },
+  delete: (resourceName: string, appName: string, deleteDisk?: boolean, signal?: AbortSignal): Promise<void> => {
```

Comment:
Hmm... resourceName and appName don't make semantic sense in the context of a disk delete. I'm not entirely sure this is the correct interface to expose. The first argument is either a workspaceId or a googleProject, the former (workspaceId) corresponding to the v2 delete endpoint and the latter (googleProject) corresponding to a v1 delete endpoint. Its similar to runtimes, where we can decide based off the resource which one to call

Reply:
hmmm... This is app delete, with disk delete being optional and driven by the existing ajax endpoint's ability to accept the deleteDisk boolean.

Reply:
ok, from our chat, the args should be (app:AppBasics, deleteDisk? boolean), 

type AppBasics = Pick<App['CloudContext'], 'cloudResouce'>, and Pick <App, 'appName'>.  
also will probably need workspaceId to determine platform for throwing 'not-supported' error.

Reply:
Nit: Instead of a boolean parameter, consider an options object or having the caller call delete app and delete disk separately.

https://alexkondov.com/should-you-pass-boolean-to-functions/
https://www.codereadability.com/what-are-javascript-options-objects/

Reply:
I do favor named argument objects.... for all the self-documenting reasons noted in that first article... tho Typescript and modern IDE hinting, etc. do take some of the edge off with boolean args.

named args are also nice for much easier optionals where order of the args is no longer an artificial constraint on things.    I am a little hesitant to introduce an args object for just one boolean tho... I usually wait until there's at least a couple optionals, bools, or the true/false is not enough so use an enum instead scenario.  Because named args objects do tend to add weight to the caller side code.  

Because so many of these data-provider methods have signal? as last optional arg, I think we would want to run this pattern by the broader community..... do we want to have all methods like this take a single args object with all args named?  Or do we have (mainObject, optionsArgs) type thing for scenarios like this where there is a disk, or app object that is the main thing being acted on?

Reply:
I find that there is almost always a better alternative to a boolean parameter.

Even when there's only one, like in this case, it's usually tangential to the main purpose of the function. And thus, it's difficult to understand what it means when reading the call. `appProvider.delete(appId)` is clear, but with `appProvider.delete(appId, true)` I have to consult the docstring (if there is one) or code to find out what that boolean is doing. An options object makes it much more clear: `appProvider.delete(appId, { deleteDisk: true })`.

Related to that, because it is also tangential to the focus of the function, I also favor putting `signal` in an options object (and have done so in [FileBrowserProvider](https://github.com/DataBiosphere/terra-ui/blob/dev/src/libs/ajax/file-browser-providers/FileBrowserProvider.ts)).

Reply:
I very much agree the readability benefits greatly... I saw some nice efficiencies in how you implemented the signatures too, which I will borrow some of... gonna give this a whirl...

Reply:
@nawatts take a look at the new provider methods signatures :)

Reply:
Nice!